foxtrot-core/src/test/java/com/flipkart/foxtrot/core/querystore/actions/FilterActionTest.java

```diff
@@ -659,6 +677,35 @@ public void testQueryCaching() throws QueryStoreException, JsonProcessingExcepti
         compare(documents, actualResponse.getDocuments());
     }
 
+    @SuppressWarnings("deprecation")
+    @Test
+    public void testMissingIndicesQuery() throws QueryStoreException {
+        List<Document> documents = TestUtils.getQueryDocumentsDifferentDate(mapper, new Date(2014 - 1900, 4, 1).getTime());
+        documents.addAll(TestUtils.getQueryDocumentsDifferentDate(mapper, new Date(2014 - 1900, 4, 5).getTime()));
+        queryStore.save(TestUtils.TEST_TABLE_NAME, documents);
+        for (Document document : documents) {
+            elasticsearchServer.getClient().admin().indices()
+                    .prepareRefresh(ElasticsearchUtils.getCurrentIndex(TestUtils.TEST_TABLE_NAME, document.getTimestamp()))
+                    .setForce(true).execute().actionGet();
+        }
+        GetIndexResponse response = elasticsearchServer.getClient().admin().indices().getIndex(new GetIndexRequest()).actionGet();
+        assertEquals(3, response.getIndices().length);
+
+        Query query = new Query();
+        query.setLimit(20);
+        query.setTable(TestUtils.TEST_TABLE_NAME);
+        BetweenFilter betweenFilter = new BetweenFilter();
+        betweenFilter.setField("_timestamp");
+        betweenFilter.setFrom(documents.get(0).getTimestamp());
+        betweenFilter.setTo(documents.get(documents.size() -1).getTimestamp());
+        betweenFilter.setTemporal(true);
+        query.setFilters(Lists.<Filter>newArrayList(betweenFilter));
+        
+        QueryResponse actualResponse = QueryResponse.class.cast(queryExecutor.execute(query));
+        assertEquals(18, actualResponse.getDocuments().size());
```

Comment:
actually it would be just documents.size(). I do a addAll to that list the the very start.